pkg/apis/v1beta1/nodepool.go

```diff
@@ -222,30 +239,45 @@ func (nl *NodePoolList) OrderByWeight() {
 	})
 }
 
-// MustGetAllowedDisruptions calls GetAllowedDisruptions and returns 0 if the error is not nil. This reduces the
-// amount of state that the disruption controller must reconcile, while allowing the GetAllowedDisruptions()
-// to bubble up any errors in validation.
-func (in *NodePool) MustGetAllowedDisruptions(ctx context.Context, c clock.Clock, numNodes int) int {
-	val, err := in.GetAllowedDisruptions(ctx, c, numNodes)
-	if err != nil {
-		return 0
+// GetAllowedDisruptionsByMethod returns the minimum allowed disruptions across all disruption budgets, for all disruption methods for a given nodepool
+func (in *NodePool) GetAllowedDisruptionsByReason(ctx context.Context, c clock.Clock, numNodes int) (map[DisruptionReason]int, error) {
+	allowedDisruptions := map[DisruptionReason]int{}
+	updateAllowedDisruptions := func(reason DisruptionReason, val int) {
+		if existingVal, ok := allowedDisruptions[reason]; ok {
+			allowedDisruptions[reason] = lo.Min[int]([]int{val, existingVal})
+			return
+		}
+		allowedDisruptions[reason] = val
```

Comment:
One way to make this easier on yourself from changing the existing min function is just to instantiate the map with MaxInt32 values, and then you don't have to define this function for ease of use later, since it'll just be one min call.

Reply:
Took your advice completely rewrote this function that I originally wrote in the late hours of the night.